config/core/200-roles/addressable-resolvers-clusterrole.yaml

```diff
@@ -12,6 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Use this aggregated ClusterRole when you need readonly access to "Addressables"
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: addressable-resolver
```

Comment:
We should prefix any cluster resources with `knative-serving` or something so that we don't conflict with eventing's installation.

Reply:
https://github.com/knative/eventing/blob/fc115ae011e18ca7cf7b67c8867d98afc3d09ab4/config/core/roles/addressable-resolvers-clusterrole.yaml#L16-L26

Reply:
https://github.com/knative/serving/pull/10693#discussion_r585637410

Reply:
I thought Matt was saying you need an instance of the ClusterRole for binding - which you added here.

I'm pointing out the role you added can have a unique name different from eventing - to avoid conflicts with installation/removal.

Is there something else I'm missing?

Reply:
the point is to be able to access arbitrary addressables, including ones in eventing, like Channel. I dont think a serving specifc role would accomplish this, but I may be wrong (we did have a whole conversation about this above where I pointed out the potential clash and opinion seemed to be it was required, though)!

Reply:
(see the comment starting "hm. Im not sure how well that will work. We'd then both be defining the same ClusterRole.")

Reply:
> . I dont think a serving specifc role would accomplish this, but I may be wrong (we did have a whole conversation about this above where I pointed out the potential clash and opinion seemed to be it was required, though)!

I did a quick test and I think it does work as I would expect - ie.  Having two uniquely named aggregated cluster roles will each aggregate their own rules - even if they have the same selector. 

```yaml
kind: ClusterRole
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: knative-serving-addressable-resolver
  labels:
    serving.knative.dev/release: devel
    # Labeled to facilitate aggregated cluster roles that act on Addressables.
    duck.knative.dev/addressable: "true"
rules:
- apiGroups:
  - serving.knative.dev
  resources:
  - routes
  - routes/status
  - services
  - services/status
  verbs:
  - get
  - list
  - watch
 ---
 apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: imc-addressable-resolver
  labels:
    eventing.knative.dev/release: devel
    duck.knative.dev/addressable: "true"
rules:
  - apiGroups:
      - messaging.knative.dev
    resources:
      - inmemorychannels
      - inmemorychannels/status
    verbs:
      - get
      - list
      - watch
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: addressable-resolver
  labels:
    eventing.knative.dev/release: devel
aggregationRule:
  clusterRoleSelectors:
  - matchLabels:
      duck.knative.dev/addressable: "true"
rules: [] # Rules are automatically filled in by the controller manager.
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: serving-addressable-resolver
  labels:
    eventing.knative.dev/release: devel
aggregationRule:
  clusterRoleSelectors:
  - matchLabels:
      duck.knative.dev/addressable: "true"
rules: [] # Rules are automatically filled in by the controller manager.
---
```

Ends up apply the same rules to both aggregated cluster roles

```yaml
aggregationRule:
  clusterRoleSelectors:
  - matchLabels:
      duck.knative.dev/addressable: "true"
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"aggregationRule":{"clusterRoleSelectors":[{"matchLabels":{"duck.knative.dev/addressable":"true"}}]},"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRole","metadata":{"annotations":{},"labels":{"eventing.knative.dev/release":"devel"},"name":"serving-addressable-resolver"},"rules":[]}
  creationTimestamp: "2021-04-16T04:20:57Z"
  labels:
    eventing.knative.dev/release: devel
    manager: kubectl-client-side-apply
    operation: Update
    time: "2021-04-16T04:20:57Z"
  name: serving-addressable-resolver
  resourceVersion: "12749001"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/serving-addressable-resolver
  uid: 24abb802-84fb-4428-b89e-b28c46408bb3
rules:
- apiGroups:
  - messaging.knative.dev
  resources:
  - inmemorychannels
  - inmemorychannels/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - serving.knative.dev
  resources:
  - routes
  - routes/status
  - services
  - services/status
  verbs:
  - get
  - list
  - watch
```

and

```yaml
aggregationRule:
  clusterRoleSelectors:
  - matchLabels:
      duck.knative.dev/addressable: "true"
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"aggregationRule":{"clusterRoleSelectors":[{"matchLabels":{"duck.knative.dev/addressable":"true"}}]},"apiVersion":"rbac.authorization.k8s.io/v1","kind":"ClusterRole","metadata":{"annotations":{},"labels":{"eventing.knative.dev/release":"devel"},"name":"addressable-resolver"},"rules":[]}
  creationTimestamp: "2021-04-15T14:33:47Z"
  labels:
    eventing.knative.dev/release: devel
  name: addressable-resolver
  resourceVersion: "12749003"
  selfLink: /apis/rbac.authorization.k8s.io/v1/clusterroles/addressable-resolver
  uid: d8e695ec-5b54-4671-9f0c-b37e62f44e4c
rules:
- apiGroups:
  - messaging.knative.dev
  resources:
  - inmemorychannels
  - inmemorychannels/status
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - serving.knative.dev
  resources:
  - routes
  - routes/status
  - services
  - services/status
  verbs:
  - get
  - list
  - watch
```

Reply:
Hm, well, works for me - agree that a duplicated role seems better than a colliding role. I don't know why we didn't spot that option in the earlier thread, but I like it: let me update and push

Reply:
done, the name is a bit unfortunate since it's prefixed `knative-serving` even though it aggregates both serving and eventing roles but eventing already took `addressable-resolver` so 🤷 .